Meta/Lagom/CMakeLists.txt

```
@@ -1,45 +1,66 @@
-cmake_minimum_required (VERSION 3.0)
-project (Lagom)
+cmake_minimum_required (VERSION 3.16)
+
+project(
+    Lagom
+    DESCRIPTION "Host build of SerenityOS libraries and Applications"
+    HOMEPAGE_URL "https://github.com/SerenityOS/serenity"
+    LANGUAGES C CXX
+)
 
 # This is required for CMake (when invoked for a Lagom-only build) to
 # ignore any files downloading during the build, e.g. UnicodeData.txt.
 # https://cmake.org/cmake/help/latest/policy/CMP0058.html
 cmake_policy(SET CMP0058 NEW)
 
-include(../CMake/wasm_spec_tests.cmake)
+get_filename_component(
+    SERENITY_PROJECT_ROOT "${PROJECT_SOURCE_DIR}/../.."
+    ABSOLUTE CACHE
+)
 
-if (NOT ENABLE_OSS_FUZZ)
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-unknown-warning-option -Wno-literal-suffix -O2 -Wall -Wextra -Werror -std=c++2a -fPIC -g -Wno-deprecated-copy")
-else()
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++2a -fPIC -g -Wno-deprecated-copy")
-endif()
+find_package(Threads REQUIRED)
+
+include(${SERENITY_PROJECT_ROOT}/Meta/CMake/wasm_spec_tests.cmake)
+
+add_compile_options(-Wno-unknown-warning-option -Wno-literal-suffix -Wno-deprecated-copy)
+add_compile_options(-O2)
+add_compile_options(-Wall -Wextra -Werror)
+add_compile_options(-fPIC -g)
+
+set(CMAKE_CXX_STANDARD 20)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_CXX_EXTENSIONS OFF)
+
+set(CMAKE_SKIP_BUILD_RPATH FALSE)
+set(CMAKE_BUILD_WITH_INSTALL_RPATH FALSE)
+set(CMAKE_INSTALL_RPATH "$ORIGIN")
```

I forgot to mention that `$ORIGIN` is Linux specific. You need a different variable for macOS machines:

```suggestion
set(CMAKE_INSTALL_RPATH "$ORIGIN")
if(APPLE)
  set(CMAKE_INSTALL_RPATH "@loader_path")
endif()
```

---

Just as a note, but I would like someone with a mac machine to chime in regarding how the runtime loader works on that system. I see that mac's runtime loader has similar special variables in the RPATH, like `@loader_path`, but I'm unsure if it's actually needed here.

---

Sounds good, I've added the apple check. Not sure what to do about the mac libraries, but I'm sure we'll find out when someone with a Mac wants to use the libjs-test262-runner after we've updated it for these changes.